### DISCUSSION TITLE
Adding different media types to an embed

### DISCUSSION THREAD

**Discussion by RageMellon** [DSharpPlus/DSharpPlus#1191] [Q&A]
Hi everyone, I've been using DSharpPlus for a while now, and have run into a problem I can find in the documentation:

When posting a range of different links into the discord chat, discord automatically creates an embed with the content of the link, like so:
![image](https://user-images.githubusercontent.com/49679541/149679657-357605bd-154f-492a-a941-e9d3bc53f124.png)

However, when trying to replicate this behaviour, whenever I load any form of content into an embed that _isn't_ an image, I get the "failure poop":

![image](https://user-images.githubusercontent.com/49679541/149679713-751ad8dc-f936-4bd7-a768-ad680d86c98e.png)

Is there something I've missed for how to load videos or GIFs into an embed via a link?
I've even noticed the `DiscordEmbedVideo` class, which can be populated by discord, but not by me?

**Comment by JulianusIV** [comment]
While these embeds for websites like YouTube exist, [Discord currently only supports sending image embeds](https://discord.com/developers/docs/reference#editing-message-attachments-using-attachments-within-embeds)
However GIFs should be entirely possible by adding them to the attachments and setting ImageURL on the embed to ``"attachment://filename.gif"``

the Video class is, as you said, only populated by Discord, and cannot be sent, therefore everything to do with that is readonly
